csharp/src/Drivers/Apache/Spark/SparkDatabase.cs

```diff
@@ -28,9 +30,15 @@ public SparkDatabase(IReadOnlyDictionary<string, string> properties)
             this.properties = properties;
         }
 
-        public override AdbcConnection Connect(IReadOnlyDictionary<string, string>? properties)
+        public override AdbcConnection Connect(IReadOnlyDictionary<string, string>? options)
         {
-            SparkConnection connection = new SparkConnection(this.properties);
+            // connection options takes precedence over database properties for the same option
+            IReadOnlyDictionary<string, string> mergedProperties = options == null
+                ? properties
+                : options
+                    .Concat(properties.Where(x => !options.Keys.Contains(x.Key, StringComparer.OrdinalIgnoreCase)))
+                    .ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
+            SparkConnection connection = SparkConnectionFactory.NewConnection(mergedProperties); // new SparkConnection(mergedProperties);
```

Comment:
The `adbc.spark.type` parameter is now required and used to determine which Spark variant connection will be used.